server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -6100,9 +6101,18 @@ public UserVm restoreVMInternal(Account caller, UserVmVO vm, Long newTemplateId)
                     throw ex;
                 }
             }
-            TemplateDataStoreVO tmplStore = _templateStoreDao.findByTemplateZoneReady(template.getId(), vm.getDataCenterId());
-            if (tmplStore == null) {
-                throw new InvalidParameterValueException("Cannot restore the vm as the template " + template.getUuid() + " isn't available in the zone");
+
+            TemplateDataStoreVO tmplStore;
```

Comment:
please, extract this bit as a separate private method, i.e. checkRestorePossible().